lib/beacon_web/live/page_live.ex

```diff
@@ -37,6 +41,17 @@ defmodule BeaconWeb.PageLive do
       """
   end
 
+  defp choose_template(templates) do
```

Comment:
You can return early here:

```suggestion
  defp choose_template(%{variants: [], primary: primary}), do: primary
  
  defp choose_template(templates) do
```

Reply:
Yes, good idea.  Here we should take every opportunity to maximize performance since it will happen on every page render.